python/ray/train/data_parallel_trainer.py

```diff
@@ -512,3 +513,24 @@ def _load_checkpoint(
         )
     model = checkpoint_dict[MODEL_KEY]
     return model, preprocessor
+
+
+def _load_checkpoint_dir(
+    checkpoint: Checkpoint, get_model_func: Callable
```

Comment:
Can you add more information on what the argument to get_model_func should be?

Tbh, having this as a separate function might be overkill given that the model_func already has to be supplied by the caller anyways.